src/MatrixFields/var_and_var_tree.jl

```diff
@@ -0,0 +1,157 @@
+"""
+    VariableName(name_chain...)
+
+A singleton type that represents a chain of `getproperty` calls, which can be
+used to access a property or sub-property of an object `x` using the function
+`get_var(x, var)`. The entire object `x` can also be accessed with the empty
+`VariableName()`.
+"""
+struct VariableName{name_chain} end
+VariableName() = VariableName{()}() # This is required for type stability.
+VariableName(name_chain...) = VariableName{name_chain}()
```

Comment:
It looks like this largely supplants the stuff in https://github.com/CliMA/ClimaCore.jl/blob/5215d8dd3496bc01db7f0aa53fd2c06b61a19a99/src/Fields/field_iterator.jl

@charleskawczynski can we combine it?

Reply:
Yes I think `field_iterator` can be replaced by (or turned into a wrapper of) `FieldVectorView`, and `property_chains` can be replaced by `top_level_vars`.

Reply:
would it make sense to move it there?

Reply:
I also don't see any tests or examples on how to use it?

Reply:
Agreed, and yes, we should definitely add tests

Reply:
Sure, I can move it and add more docs/tests. Two questions, though:
- Should we try to maintain the old API of `property_chains` and `single_field` (and just make those functions into wrappers for the new functions), or should we make this a breaking change and have users switch to writing `for field in FieldVectorView(Y) ... end`?
- The `VariableName` infrastructure (`VarTree` + `VarSet` + `VarDict`) depends on both `Field`s and `ColumnwiseBandMatrixField`s, the latter of which is only defined in the MatrixFields module. Should we move the `ColumnwiseBandMatrixField` definition to the Fields module (and maybe move all the matrix multiplication code to the Operators module)? If so, maybe that can be a future PR?